### DISCUSSION TITLE
Where do i add samplers?

### DISCUSSION THREAD

**Discussion by ArmoredCreeper97** [Comfy-Org/ComfyUI#1205] [Q&A]
I want to use a sampler that isnt included. Which folder do i add it into?

**Comment by alelordelo** [comment]
Also interested in this

**Comment by blepping** [comment]
you'd basically need to adapt the sampler into a ComfyUI extension. one way to do it is to add a node that returns a `SAMPLER` which can be used with the built in `SamplerCustom` node. it's also possible to mess with the built in list and make it show up in the built in samplers (so you don't need to use `SamplerCustom`).

here are some examples of extensions that add samplers:

1. https://github.com/blepping/ComfyUI-sonar
2. https://github.com/Clybius/ComfyUI-Extra-Samplers

you can also look at `comfy_extras/nodes_custom_sampler.py` in the comfyui repo, it adds some nodes for samplers (just wrappers for built in ones though).
